src/v/cloud_storage/tests/directory_walker_test.cc

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "bytes/iobuf.h"
+#include "cloud_storage/recursive_directory_walker.h"
+#include "seastarx.h"
+
+#include <seastar/core/file.hh>
+#include <seastar/core/fstream.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include <chrono>
+
+using namespace cloud_storage;
+using namespace std::chrono_literals;
+
+SEASTAR_THREAD_TEST_CASE(one_level) {
+    cloud_storage::recursive_directory_walker _walker;
+    const std::filesystem::path target_dir{"target_dir"};
+    const std::filesystem::path file_path1{"target_dir/file1.txt"};
+    const std::filesystem::path file_path2{"target_dir/file2.txt"};
+
+    ss::recursive_touch_directory(target_dir.native()).get();
+
+    auto flags = ss::open_flags::wo | ss::open_flags::create
+                 | ss::open_flags::exclusive;
+    auto file1 = ss::open_file_dma(file_path1.native(), flags).get();
+    file1.close().get();
+
+    ss::sleep(ss::lowres_clock::duration(1s)).get();
```

Comment:
Why do we need sleeps? Unit tests shouldn't introduce unnecessary delays.

Reply:
cause `directory_walker` returns files sorted by creation timestamp, this sleeps are needed to make the order stable.

Reply:
Got it, I think it merits a comment.

Reply:
Added comment!